docs/man/git-lfs-config.adoc

```diff
@@ -318,6 +318,13 @@ Supports URL config lookup as described in:
 https://git-scm.com/docs/git-config#git-config-httplturlgt. To set this
 value per-host:
 `git config --global lfs.https://github.com/.locksverify [true|false]`.
+* `lfs.<url>.sshprotocol`
```

Comment:
As it happens, I've been slowly working on some revised and expanded documentation for many of our config settings, and in the process have written some for the currently undocumented `lfs.gitProtocol` setting, which takes values like `http` or `https`; the intent is that whatever is specified (with the default being `https`) is then substituted if the original URL begins with the `git://` scheme.  So it's a full protocol substitution.

This new setting has a different intent, isn't to change from the SSH protocol to something else entirely, but rather to adjust "how much" SSH to use, as it were.  So I wonder if we could use a different term here other than "protocol", although I appreciate the challenge in coming up with something clear and succinct.

One alternative which occurred to me is something like `lfs.sshTransferProtocol` or `lfs.sshAdapterProtocol`; another, which I maybe like better, might be `lfs.sshTransfer`, which aligns with the `lfs.customTransfer.*` settings.

Following from that last idea, I then wondered if it might be reasonable to allow as the possible values some simpler terms such as `always`, `never`, and `negotiate` (the default), with the goal of making the configuration settings more readable to the lay user.

I also toyed with the idea of having two Boolean settings instead of a single ternary one, maybe something like:
- `lfs.sshTransfer` - if false, `git-lfs-authenticate` is always used; otherwise, `git-lfs-transfer` may be used (default: true)
- `lfs.sshTransfer.negotiate` - only meaningful if `lfs.sshTransfer` is true, in which case it indicates whether or not to try `git-lfs-transfer` before falling back to `git-lfs-authenticate` (default: true)

That would avoid the need to come up with any terms at all for the settings, which I kind of appreciate, because naming things is hard.

Reply:
On reflection, `lfs.sshTransfer.negotiate` is a bit ambiguous and maybe suggests some kind of negotiation of the underlying SSH protocol or the LFS transfer protocol over SSH, and neither is the case (particularly not the former).

What about something like `lfs.sshTransfer.httpFallback` or `lfs.sshTransfer.basicTransferFallback`, which might make it really clear what's being enabled/disabled by a Boolean value?

Reply:
I think having a single tri-state option is better here.  I'll take `lfs.sshTransfer` with `always`, `never`, and `negotiate`.

Reply:
Sounds good, thanks!